cJSON.c

```diff
@@ -1137,7 +1156,11 @@ static char *print_array(const cJSON *item, int depth, cjbool fmt, printbuffer *
         child = item->child;
         while (child && !fail)
         {
-            print_value(child, depth + 1, fmt, p);
+            ptr = print_value(child, depth + 1, fmt, p);
+            if (!ptr)
+            {
+                return NULL;
+            }
```

Comment:
Change this to:
```c
if (!print_value(child, depth + 1, fmt, p))
{
    return NULL;
}
```

Thank you for finding this problem btw. after that I was taking a closer look to the rest of the library and there are quite some cases where out of memory errors are not handled properly.